Crash report

Reported here: https://support.andbible.org/scp/tickets.php?id=516

Device info:
And Bible version: 3.3.400#6a52aa2ba (built 29/04/21 20:09:21)
Android version: 7.1.1
Android SDK version: 25
Manufacturer: samsung
Model: SAMSUNG-SM-T377A

--------- beginning of crash
05-15 16:55:57.783  7299  7299 E AndroidRuntime: FATAL EXCEPTION: main
05-15 16:55:57.783  7299  7299 E AndroidRuntime: Process: net.bible.android.activity, PID: 7299
05-15 16:55:57.783  7299  7299 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=4, result=0, data=null} to activity {net.bible.android.activity/net.bible.android.view.activity.page.MainBibleActivity}: java.lang.NullPointerException
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4556)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4599)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.ActivityThread.-wrap22(ActivityThread.java)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1715)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6823)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1557)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: Caused by: java.lang.NullPointerException
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at net.bible.android.view.activity.page.MainBibleActivity.onActivityResult(MainBibleActivity.kt:1217)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:7322)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4552)
05-15 16:55:57.783  7299  7299 E AndroidRuntime: 	... 9 more

--- a/app/src/main/java/net/bible/android/view/activity/page/MainBibleActivity.kt
+++ b/app/src/main/java/net/bible/android/view/activity/page/MainBibleActivity.kt
@@ -1289,90 +1289,94 @@ class MainBibleActivity : CustomTitlebarActivityBase() {
 
     public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         Log.d(TAG, "Activity result:$resultCode")
-        when(requestCode) {
-            WORKSPACE_CHANGED -> {
-                val extras = data?.extras
-                val workspaceId = extras?.getLong("workspaceId")
-                val changed = extras?.getBoolean("changed")
+        val extras = data?.extras
+        if (extras != null) {
+            when (requestCode) {
+                WORKSPACE_CHANGED -> {
+                    val workspaceId = extras.getLong("workspaceId")
+                    val changed = extras.getBoolean("changed")
 
-                if(resultCode == Activity.RESULT_OK) {
-                    if(workspaceId != 0L && workspaceId != currentWorkspaceId) {
-                        currentWorkspaceId = workspaceId!!
-                    } else if(changed == true) {
-                        currentWorkspaceId = currentWorkspaceId
+                    if (resultCode == Activity.RESULT_OK) {
+                        if (workspaceId != 0L && workspaceId != currentWorkspaceId) {
+                            currentWorkspaceId = workspaceId
+                        } else if (changed) {
+                            currentWorkspaceId = currentWorkspaceId
+                        }
                     }
-                }
-                return
-            }
-            COLORS_CHANGED -> {
-                val extras = data?.extras!!
-                val edited = extras.getBoolean("edited")
-                val reset = extras.getBoolean("reset")
-                val windowId = extras.getLong("windowId")
-                val colorsStr = extras.getString("colors")
-
-                if(!edited && !reset) return
-
-                val colors = if(reset)
-                    if(windowId != 0L) {
-                        null
-                    } else TextDisplaySettings.default.colors
-                else
-                    WorkspaceEntities.Colors.fromJson(colorsStr!!)
-
-                if(windowId != 0L) {
-                    val window = windowRepository.getWindow(windowId)!!
-                    window.pageManager.textDisplaySettings.colors = colors
-                    window.bibleView?.updateTextDisplaySettings()
-                } else {
-                    windowRepository.textDisplaySettings.colors = colors
-                    windowRepository.updateWindowTextDisplaySettingsValues(setOf(TextDisplaySettings.Types.COLORS), windowRepository.textDisplaySettings)
-                    windowRepository.updateAllWindowsTextDisplaySettings()
-                }
-                resetSystemUi()
-                invalidateOptionsMenu()
-            }
-            TEXT_DISPLAY_SETTINGS_CHANGED -> {
-                val extras = data?.extras!!
-
-                val edited = extras.getBoolean("edited")
-                val reset = extras.getBoolean("reset")
-
-                val settingsBundle = SettingsBundle.fromJson(extras.getString("settingsBundle")!!)
-                val requiresReload = extras.getBoolean("requiresReload")
-
-                if(!edited && !reset) return
-
-                val dirtyTypes = DirtyTypesSerializer.fromJson(extras.getString("dirtyTypes")!!).dirtyTypes
-
-                workspaceSettingsChanged(settingsBundle, requiresReload, reset, dirtyTypes)
-                return
-            }
-            STD_REQUEST_CODE -> {
-                val classes = arrayOf(GridChoosePassageBook::class.java.name, Bookmarks::class.java.name)
-                val className = data?.component?.className
-                if (className != null && classes.contains(className)) {
-                    val isFromBookmark = className == Bookmarks::class.java.name
-                    val verseStr = data.extras!!.getString("verse")
-                    val verse = try {
-                        VerseFactory.fromString(navigationControl.versification, verseStr)
-                    } catch (e: NoSuchVerseException) {
-                        ABEventBus.getDefault().post(ToastEvent(getString(R.string.verse_not_found)))
-                        return
-                    }
-                    val pageManager = windowControl.activeWindowPageManager
-                    if(isFromBookmark && !pageManager.isBibleShown) {
-                        pageManager.setCurrentDocumentAndKey(windowControl.defaultBibleDoc(false), verse)
-                    } else
-                        pageManager.currentPage.setKey( verse)
                     return
                 }
-            }
-            IntentHelper.UPDATE_SUGGESTED_DOCUMENTS_ON_FINISH -> {
-                updateDocuments()
-                return
+                COLORS_CHANGED -> {
+                    val edited = extras.getBoolean("edited")
+                    val reset = extras.getBoolean("reset")
+                    val windowId = extras.getLong("windowId")
+                    val colorsStr = extras.getString("colors")
+
+                    if (!edited && !reset) return
+
+                    val colors = if (reset)
+                        if (windowId != 0L) {
+                            null
+                        } else TextDisplaySettings.default.colors
+                    else
+                        WorkspaceEntities.Colors.fromJson(colorsStr!!)
+
+                    if (windowId != 0L) {
+                        val window = windowRepository.getWindow(windowId)!!
+                        window.pageManager.textDisplaySettings.colors = colors
+                        window.bibleView?.updateTextDisplaySettings()
+                    } else {
+                        windowRepository.textDisplaySettings.colors = colors
+                        windowRepository.updateWindowTextDisplaySettingsValues(
+                            setOf(TextDisplaySettings.Types.COLORS),
+                            windowRepository.textDisplaySettings
+                        )
+                        windowRepository.updateAllWindowsTextDisplaySettings()
+                    }
+                    resetSystemUi()
+                    invalidateOptionsMenu()
+                }
+                TEXT_DISPLAY_SETTINGS_CHANGED -> {
+                    val edited = extras.getBoolean("edited")
+                    val reset = extras.getBoolean("reset")
+
+                    val settingsBundle = SettingsBundle.fromJson(extras.getString("settingsBundle")!!)
+                    val requiresReload = extras.getBoolean("requiresReload")
+
+                    if (!edited && !reset) return
+
+                    val dirtyTypes = DirtyTypesSerializer.fromJson(extras.getString("dirtyTypes")!!).dirtyTypes
+
+                    workspaceSettingsChanged(settingsBundle, requiresReload, reset, dirtyTypes)
+                    return
+                }
+                STD_REQUEST_CODE -> {
+                    val classes = arrayOf(GridChoosePassageBook::class.java.name, Bookmarks::class.java.name)
+                    val className = data.component?.className
+                    if (className != null && classes.contains(className)) {
+                        val isFromBookmark = className == Bookmarks::class.java.name
+                        val verseStr = extras.getString("verse")
+                        val verse = try {
+                            VerseFactory.fromString(navigationControl.versification, verseStr)
+                        } catch (e: NoSuchVerseException) {
+                            ABEventBus.getDefault().post(ToastEvent(getString(R.string.verse_not_found)))
+                            return
+                        }
+                        val pageManager = windowControl.activeWindowPageManager
+                        if (isFromBookmark && !pageManager.isBibleShown) {
+                            pageManager.setCurrentDocumentAndKey(windowControl.defaultBibleDoc(false), verse)
+                        } else
+                            pageManager.currentPage.setKey(verse)
+                        return
+                    }
+                }
             }
         }
+
+        if (requestCode == IntentHelper.UPDATE_SUGGESTED_DOCUMENTS_ON_FINISH) {
+            updateDocuments()
+            return
+        }
+
         super.onActivityResult(requestCode, resultCode, data)
         when {
             mainMenuCommandHandler.restartIfRequiredOnReturn(requestCode) -> {

